staging/kos/pkg/networkfabric/interfaces.go

```diff
@@ -18,33 +18,100 @@ package networkfabric
 
 import "net"
 
-// Interface is the interface of a network fabric which allows
-// the user to implement Netowrk Interfaces. The implementer
-// MUST return nil when the user attempts to delete an Interface
-// which does not exist.
+// Interface is the contract of a VXLAN network fabric.
+// It declares functions to create Network Interfaces that are part of a VXLAN
+// segment.
+// The VXLAN segment of a Network Interface is an invocation argument of the
+// functions to create the Network Interface.
+// All traffic sent/received to/from the Network Interfaces created via an
+// implementer of this contract MUST be VXLAN-tunneled.
+//
+// This contract makes a distinction between local and remote Network Interfaces.
+//
+// A local Network Interface is the networking state of a guest that is bound to
+// the same node as the user of this contract. Creating a local Network
+// Interface means creating a Linux network device and configuring the node's
+// networking state so that the Linux network device can send/receive VXLAN-tunneled
+// traffic to/from other guests (local or remote) in its VXLAN segment.
+//
+// A remote Network Interface is the networking state of a guest that is bound
+// to a node other than that of the user of this contract. Creating a remote
+// Network Interface means configuring the networking state on the node of the
+// user of this contract so that traffic generated on such node and directed at
+// the remote Network Interface is correctly VXLAN-tunneled to the node of the
+// remote Network Interface.
+//
+// Network Interfaces are identified by a name, which must be unique over space.
+// (VNI, guest IP) pairs must also be unique over space, that is, for this
+// contract two Network Interfaces with the same (VNI, guest IP) pair are the
+// same Network Interface. Follow some guarantees that implementers MUST make:
```

Comment:
What about (VNI, GuestMAC)?

Reply:
On the basis that a VNI identifies a VXLAN segment, and a VXLAN segment represents a Layer2 network, where IDs are MAC addresses, as opposed to IP addresses?

Reply:
yes

Reply:
Sounds good, will change it tomorrow.